conda_build/metadata.py

```diff
@@ -279,15 +279,18 @@ def check_fields(meta, strictness=.9):
     try:
         _ = _meta['package']['name']
         _ = _meta['package']['version']
-    except IndexError:
-        raise ValueError("Invalid recipe")
+    except KeyError:
+        sys.exit("Invalid meta.yaml: Missing package/name or package/version.")
```

Comment:
I'd prefer an exception being thrown.  sys.exit shouldn't be used anywhere under the user interface.